### DISCUSSION TITLE
How do I make the text in a CollapsingHeader wrap?

### DISCUSSION THREAD

**Discussion by zkldi** [emilk/egui#2976] [Q&A]
I have a `ui` that i'd like to keep around `x=400`, and it has a bunch of collapsing headers in it. The wrapping APIs seem to not affect CollapsingHeaders.

Is there a nice way to make these things wrap their text if they exceed a certain width?

**Comment by higumachan** [comment]
> Is there a nice way to make these things wrap their text if they exceed a certain width?

If there is no problem with truncate instead of wrap, the following code can be used to create it.

```rust
CollapsingState::load_with_default_open(
                        ui.ctx(),
                        id,
                        false,
                    )
                    .show_header(ui, |ui| {
                            Label::new(name).truncate(true).ui(ui)
                    })
                    .body(|ui| {
                        ... your collapsing contents
                    });
```

I've never done it, but if you do it right, you may be able to handle the turnaround!
